### PR TITLE
let's make it so npm search on npmo can search for scoped modules

### PR DESCRIPTION
**Pull request by bcoe**
* Pull in the `npmo` branch of npm-registry-couchapp, allowing CLI search for scoped modules.
* add a new configuration option so that npm On-Site users can lock down this search.

### CODE REVIEW

File: replicated/replicated.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ admin_commands:
   component: npme
   image:
     image_name: npme
-    version: '1.0.22'
+    version: '1.0.24'
```
Comment (nexdrew) [inline replicated/replicated.yml]: Are these versions arbitrary?

File: replicated/replicated.yml [hunk 2]
```diff
@@ -723,6 +726,24 @@ config:
...
     required: false
+  - name: scoped_search
+    title: Scoped CLI Search
+    description: Should npm search be enabled for scoped modules?
+    items:
+    - name: scoped_search
+      type: select_one
+      default: scoped_search_yes
+      items:
+      - name: scoped_search
```
Comment (nexdrew) [inline replicated/replicated.yml]: This should be `scoped_search_no`, correct?

File: replicated/replicated.yml [hunk 3]
```diff
@@ -723,6 +726,24 @@ config:
...
+        title: No
+        type: text
+        affix: left
+        required: false
+      - name: scoped_search
```
Comment (nexdrew) [inline replicated/replicated.yml]: And this `scoped_search_yes`?

File: replicated/replicated.yml (hunk 1 above)
Comment (bcoe) [inline replicated/replicated.yml]: They're somewhat arbitrary, they're what the tags get set as in our Docker registries -- I forgot to push an update with `1.0.23`, which is why it jumped to `24.

File: replicated/Dockerfile.replicated [hunk 4]
```diff
@@ -8,7 +8,7 @@ COPY ./replicated/start.sh /etc/npme/start.sh
 WORKDIR /etc/npme
 
 RUN echo '@npm:registry=https://enterprise.npmjs.com/' > ~/.npmrc
-RUN npm install npm-registry-couchapp
+RUN npm install npm-registry-couchapp@npmo
```
Comment (nexdrew) [inline replicated/Dockerfile.replicated]: Is the `npmo` tag a long-term solution? Will need to make sure no one accidentally removes it. :)

**Comment by nexdrew** [comment]
LGTM :+1:

File: replicated/Dockerfile.replicated (hunk 4 above)
Comment (bcoe) [inline replicated/Dockerfile.replicated]: Definitely not a great long term solution see:

https://github.com/npm/issue-tracker/issues/294

**Pull request merged**